include/cubos/settings.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef CUBOS_SETTINGS_HPP
+#define CUBOS_SETTINGS_HPP
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace cubos
+{
+
+    class Settings
+    {
+    protected:
+        Settings()

```

Comment:
Should be changed to Settings() = default; This way it becomes shorter and there's no need to move it to the source file.